src/classes/BDI_ObjectMappingLogic.cls

```diff
@@ -0,0 +1,92 @@
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE 
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, 
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, 
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT 
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN 
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2019
+* @group Batch Data Import
+* @group-content ../../ApexDocContent/BatchDataImport.htm
+* @description Virtual class containing the standard logic for mapping fields between source and target objects.  Can be overriden by clients
+* who need custom mapping logic.
+*/
+
+public with sharing virtual class BDI_ObjectMappingLogic {
+
+    public virtual BDI_ObjectWrapper[] populateObjects(BDI_ObjectWrapper[] objWraps) {
+        if (objWraps == null) {
+            return objWraps;
+        }
+
+        for (BDI_ObjectWrapper objWrap : objWraps) {
+            Map<String, Schema.DescribeFieldResult> fieldDescribeMap = UTIL_Describe.getAllFieldsDescribe(UTIL_Namespace.adjustNPSPNamespace(objWrap.objMapping.Object_API_Name__c));
+
+            objWrap.sObj = Schema.getGlobalDescribe().get(UTIL_Namespace.adjustNPSPNamespace(objWrap.objMapping.Object_API_Name__c)).newSObject();
+
+            // If the object already has an existing sobject Id, then set it in the sObj
+            if (objWrap.existingSobjectId != null) {
+                objWrap.sObj.put('Id', objWrap.existingSobjectId);
+            }
+
+            updateChildToPredecessorRelationship(objWrap);
+            
+            for (Data_Import_Field_Mapping__mdt fieldMapping : objWrap.fieldMappings) {
+                String sourceFieldName = UTIL_Namespace.adjustNPSPNamespace(fieldMapping.Source_Field_API_Name__c);
+                String targetFieldName = UTIL_Namespace.adjustNPSPNamespace(fieldMapping.Target_Field_API_Name__c);
+                
+                Schema.DescribeFieldResult targetDescribe = fieldDescribeMap.get(targetFieldName.toLowerCase());
+
+                //Confirm that it is real field, and that it is accessible to the running user.
+                if (targetDescribe != null && targetDescribe.isAccessible()) {
+
+                    //Confirm that either the sObj Id is null (ie new record) or that the field is updatable if it is not new.
+                    if (objWrap.sObj.Id == null || targetDescribe.isUpdateable()) {
+
+                        Object value = objWrap.di.get(sourceFieldName);
+
+                        //only copy values that are not null to avoid overwriting data when updating existing sobjects.
+                        if (value != null) {
```

Comment:
Do we need to include the logic to map convert Boolean from DI Picklist to support checkbox fields?

Reply:
I've added a version of the logic that was in the DataImportService to do this.